### ISSUE TITLE
Question regarding the output of the adversary.

### ISSUE THREAD

**Issue by ando-khachatryan** [jirenz/HiDDeN#1]
Hi, 
First of all - very interesting paper, and many thanks for sharing the code.

I have a question regarding the adversarial part of the network. 
Quoting the paper, 
"The adversary has a structure similar to the decoder but has fewer convolutions.
It contains 3 Conv-BN-ReLU blocks with 64 filters each. The activation volume is averaged over spatial dimensions and a linear layer with two output units produces the logits for the two-class classification problem."
My question is: why linear layer with two outputs? Why not a sigmoid? I looked at the code and you even have a **commented-out** line in the adversary.lua which adds a sigmoid to the network, so I assume this is on purpose and is better than having a sigmoid, right?

A related question is -- how does it train with CrossEntropy loss and a linear activation? If the output of the last layer is negative, can the network calculate a loss and do a backprop?

Thank you once again!

**Comment by jirenz** [comment]
We are using [nn.CrossEntropyCriterion](https://github.com/torch/nn/blob/master/doc/criterion.md#nn.CrossEntropyCriterion) on adversary outputs. This loss does both log softmax and cross entropy (to speed up computation) so we don't need to normalize network output.

**Comment by ando-khachatryan** [comment]
Thanks a lot!
